src/_reduceLazy.ts

```diff
@@ -25,14 +25,17 @@ export function _reduceLazy<T, K>(
   array: Array<T>,
   lazy: (item: T, index?: number, array?: Array<T>) => LazyResult<K>,
   indexed?: boolean
-) {
-  return array.reduce((acc: Array<K>, item, index) => {
+): Array<K> {
+  const newArray: Array<K> = [];
```

Comment:
This should probably be called "ret" or "output" or "reduced" or something like that, as newArray is not really describing the semantics. But that's just my pet peeve so feel free to ignore...